nixos/modules/services/networking/ssh/sshd.nix

```diff
@@ -108,47 +140,211 @@ let
-    ));
-  in lib.listToAttrs (map mkAuthPrincipalsFile usersWithPrincipals);
+  authKeysFiles =
+    let
+      mkAuthKeyFile =
+        u:
+        lib.nameValuePair "ssh/authorized_keys.d/${u.name}" {
+          mode = "0444";
+          source = pkgs.writeText "${u.name}-authorized_keys" ''
+            ${lib.concatStringsSep "\n" u.openssh.authorizedKeys.keys}
+            ${lib.concatMapStrings (f: lib.readFile f + "\n") u.openssh.authorizedKeys.keyFiles}
+          '';
+        };
+      usersWithKeys = lib.attrValues (
+        lib.flip lib.filterAttrs config.users.users (
+          n: u:
+          lib.length u.openssh.authorizedKeys.keys != 0 || lib.length u.openssh.authorizedKeys.keyFiles != 0
+        )
+      );
+    in
+    lib.listToAttrs (map mkAuthKeyFile usersWithKeys);
+
+  authPrincipalsFiles =
+    let
+      mkAuthPrincipalsFile =
+        u:
+        lib.nameValuePair "ssh/authorized_principals.d/${u.name}" {
+          mode = "0444";
+          text = lib.concatStringsSep "\n" u.openssh.authorizedPrincipals;
+        };
+      usersWithPrincipals = lib.attrValues (
+        lib.flip lib.filterAttrs config.users.users (n: u: lib.length u.openssh.authorizedPrincipals != 0)
+      );
+    in
+    lib.listToAttrs (map mkAuthPrincipalsFile usersWithPrincipals);
 
 in
 
 {
   imports = [
-    (lib.mkAliasOptionModuleMD [ "services" "sshd" "enable" ] [ "services" "openssh" "enable" ])
-    (lib.mkAliasOptionModuleMD [ "services" "openssh" "knownHosts" ] [ "programs" "ssh" "knownHosts" ])
-    (lib.mkRenamedOptionModule [ "services" "openssh" "challengeResponseAuthentication" ] [ "services" "openssh" "kbdInteractiveAuthentication" ])
-
-    (lib.mkRenamedOptionModule [ "services" "openssh" "kbdInteractiveAuthentication" ] [  "services" "openssh" "settings" "KbdInteractiveAuthentication" ])
-    (lib.mkRenamedOptionModule [ "services" "openssh" "passwordAuthentication" ] [  "services" "openssh" "settings" "PasswordAuthentication" ])
-    (lib.mkRenamedOptionModule [ "services" "openssh" "useDns" ] [  "services" "openssh" "settings" "UseDns" ])
-    (lib.mkRenamedOptionModule [ "services" "openssh" "permitRootLogin" ] [  "services" "openssh" "settings" "PermitRootLogin" ])
-    (lib.mkRenamedOptionModule [ "services" "openssh" "logLevel" ] [  "services" "openssh" "settings" "LogLevel" ])
-    (lib.mkRenamedOptionModule [ "services" "openssh" "macs" ] [  "services" "openssh" "settings" "Macs" ])
-    (lib.mkRenamedOptionModule [ "services" "openssh" "ciphers" ] [  "services" "openssh" "settings" "Ciphers" ])
-    (lib.mkRenamedOptionModule [ "services" "openssh" "kexAlgorithms" ] [  "services" "openssh" "settings" "KexAlgorithms" ])
-    (lib.mkRenamedOptionModule [ "services" "openssh" "gatewayPorts" ] [  "services" "openssh" "settings" "GatewayPorts" ])
-    (lib.mkRenamedOptionModule [ "services" "openssh" "forwardX11" ] [  "services" "openssh" "settings" "X11Forwarding" ])
+    (lib.mkAliasOptionModuleMD
```

Comment:
IMHO this is soo much worse with the new formatter and looks horrible

Reply:
It's the official formatter. If you want to complain about it, I suggest you take it up with the `nixfmt-rfc-style` developers.